pallets/pallet-asset-switch/src/switch.rs

```diff
@@ -51,8 +92,77 @@ pub enum SwitchPairStatus {
 	Running,
 }
 
+// Constructor impls
 impl<AccountId> SwitchPairInfo<AccountId> {
-	pub(crate) fn can_switch(&self) -> bool {
+	pub(crate) fn from_input_unchecked(
+		NewSwitchPairInfo {
+			pool_account,
+			remote_asset_circulating_supply,
+			remote_asset_ed,
+			remote_asset_id,
+			remote_asset_total_supply,
+			remote_xcm_fee,
+			remote_reserve_location,
+			status,
+		}: NewSwitchPairInfo<AccountId>,
```

Comment:
I would say `input:  NewSwitchPairInfo<AccountId>` would look cleaner.

Reply:
It does not really matter, since we need to destructure the input anyway later on to access the fields. I guess this is a matter of personal preference, not really worth discussing on.